osquery/tables/events/windows/powershell_events.cpp

```diff
@@ -0,0 +1,256 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#include <math.h>
+
+#include <boost/property_tree/ptree.hpp>
+#include <boost/property_tree/xml_parser.hpp>
+
+#include <osquery/config.h>
+#include <osquery/core.h>
+#include <osquery/database.h>
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+
+#include "osquery/config/parsers/feature_vectors.h"
+#include "osquery/core/conversions.h"
+#include "osquery/core/json.h"
+#include "osquery/core/windows/wmi.h"
+#include "osquery/events/windows/windows_event_log.h"
+#include "osquery/filesystem/fileops.h"
+
+namespace pt = boost::property_tree;
+
+namespace osquery {
+
+/*
+ * Default character frequency values, derived from Lee Holmes cobbr
+ * https://gist.github.com/cobbr/acbe5cc7a186726d4e309070187beee6
+ * Note these frequencies normalize away spaces and comments, and mirror
+ * upper and lower case characters under the assumption that lower and
+ * upper case characters occur with the same frequency. It is encouraged
+ * that one pass in they're own supplied character frequency vector using
+ * the feature vector config parser.
+ */
+
+double kDefaultFrequencies[255] = {
```

Comment:
I feel like this should be either a header, or a configuration file we provide like a pack, not hardcoded. Also why a c array instead of a vector?

Reply:
I think of this vector similar to default values of strings in other places used for a filler value. I honestly wouldn't encourage folks to use this, but it's something to get started if you have no other data to pass to the parser. I don't think it belongs in a header, as I'd rather not encourage folks to bundle things up as compiled resources, so if you wanna push back on this more I'll just remove it, but I'd love to give folks something to get started looking at the similarity values.